base/shell/explorer/notifyiconscust.cpp

```diff
@@ -20,31 +20,193 @@
 
 #include "precomp.h"
 
+#define BITMAP_WIDTH 16
+#define BITMAP_HEIGHT 16
+
+#define IDC_NOTIFY_OPTIONS 5000
+
+typedef struct
+{
+    DWORD hwnd;
+    DWORD uId;
+    DWORD uCallbackMsg;
+    DWORD unkow[2];
+    HICON hIcon;
+    WCHAR text[64];
+}TRAYDATA;
+
+// Returns TRUE if successful, and FALSE otherwise. 
+BOOL AddListViewColumn(HWND hWndListView, LPWSTR label, INT iCol, int w)
+{
+    LVCOLUMNW lvc;
+
+    lvc.mask = LVCF_FMT | LVCF_WIDTH | LVCF_TEXT | LVCF_SUBITEM;
+    lvc.iSubItem = iCol;
+    lvc.pszText = label;
+    lvc.cchTextMax = lstrlenW(label);
+    lvc.cx = w;
+    lvc.fmt = LVCFMT_CENTER;
+
+    return (ListView_InsertColumn(hWndListView, iCol, &lvc) != -1);
```

Comment:
with this new implementation, you're not able to raise a ERR() in case of failure (in particular since it's very unlikely)

Reply:
See below. disregard this comment.